docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>DeleteCommand</code> object based on that user input.
+
+The <code>Shoco</code> class then calls the <code>execute</code> method of the <code>DeleteCommand</code> object
+which makes another call to the <code>deleteItem</code> function of the <code>ShoppingList</code> object 
+with the specified index.
+
+The following sequence diagram below shows how the delete feature works. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/Delete.png)
+
+#### 3.3.2 Design considerations
+
+##### Aspect: Data structure to support the delete feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>DeleteCommand</code>
+ 
+  - Pros: Easy to add the delete feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement delete feature in the <code>Shoco</code> class
+
+  - Pros: Will have less code to deal with as a new function is simply created in the <code>Shoco</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Shoco</code> class
+    simply executes those commands as black boxes, without worrying about their internal details
+
+  
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+### 3.4 Find feature
+#### 3.4.1 Current implementation
+
+The find feature is implemented using a <code>FindCommand</code> class which extends the main
+<code>Command</code> class with a String representing the keyword specified by the user.
+ 
+The <code>Shoco</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>FindCommand</code> object based on that user input.
+
+The <code>Shoco</code> class then calls the <code>execute</code> method of the <code>FindCommand</code> object
+which makes various calls to the <code>getItem</code> method of the <code>ShoppingList</code> object 
+to check whether the <code>Item</code> at each specified index contains the given keyword.
+
+Each <code>Item</code> that contains the keyword is then added to a new <code>ArrayList</code> named
+ <code>filteredItems</code> that is maintained by the <code>FindCommand</code>, which then prints the list to standard output once it is done
+creating the list.
+
+The following sequence diagram below shows how the <code>Shoco</code> object creates the <code>FindCommand</code> object. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/Find.png)
```

Comment:
![image](https://user-images.githubusercontent.com/40775612/77995103-fdd3a180-72df-11ea-9f23-eec47549e11b.png)

The sequence diagrams are great, and I have an idea of how the app works now. However, I noticed that most sequence diagrams contain how Duke and Parser work together, and the only things that change are the circled parts (FindCommand, AddCommand, DisplayCommand). Maybe you could add a separate section on how Duke and Parser work together, and then omit it from the other sequence diagrams?

Reply:
Then, the diagrams would be more focused on the feature itself